src/lib.rs

```diff
@@ -139,13 +139,35 @@ pub fn derive_pwrite(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
-fn size_with(name: &syn::Ident) -> proc_macro2::TokenStream {
+fn size_with(name: &syn::Ident, fields: &syn::FieldsNamed) -> proc_macro2::TokenStream {
+    let items: Vec<_> = fields.named.iter().map(|f| {
+        let ty = &f.ty;
+        match ty {
+            &syn::Type::Array(ref array) => {
+                let elem = &array.elem;
+                match array.len {
+                    syn::Expr::Lit(syn::ExprLit { lit: syn::Lit::Int(ref int), ..}) => {
+                        let size = int.value() as usize;
+                        quote! {
+                            (#size * <#elem>::size_with(ctx))
+                        }
+                    },
+                    _ => panic!("Pread derive with bad array constexpr")
+                }
+            },
+            _ => {
+                quote! {
+                    <#ty>::size_with(ctx)
```

Comment:
won't this require SizeWith is imported in order to use this method? (which if true means this would break everyone who hadn't incidentally imported SizeWith in that file)

Reply:
Possibly! I guess we should bring the trait into scope in the method body?

Reply:
As it turns out this works just fine without importing `SizeWith`. I can't find anything that says that this is expected behavior, but in fact if I put `use ::scroll::ctx::SizeWith;` in the `fn size_with` we're generating I get `warning: unused import: SizeWith` for it!